File: 560-subarray-sum-equals-k/subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int count = 0;
        int n = nums.length;

        for (int i = 0; i < n; i++) {
            int cursum = 0;

            for (int j = i; j < n; j++) {
                cursum += nums[j];

                if (cursum == k) {
                    count++;
                }
            }
        }

        return count;
    }
}

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        subarray ={0:1}
        cursum = 0
        count=0
        n= len(nums)

        for i in nums:
            cursum +=i

            if cursum -k in subarray:
                count+=subarray[cursum-k]

            subarray[cursum]=1+subarray.get(cursum,0)
        return count